src/rust/engine/BUILD

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+_RELEASE = env("MODE") == "release"
+_RUST_MODE = "--release" if _RELEASE else ""
+_OUTPUT_DIR = "release" if _RELEASE else "debug"
+
+files(
+    name="rust_sources",
+    sources=[
+        "Cargo.lock",
+        "build.rs",
+        "**/Cargo.toml",
+        "**/*.rs",
+        "**/*.proto",
+        ".cargo/config",
+    ],
+)
+
+shell_sources(name="shell")
+
+shell_command(
+    name="engine-and-client",
+    command=f"./sandboxed_cargo.sh build {_RUST_MODE} --features=extension-module -p engine -p client",
```

Comment:
Nit: rather than use this separate script, what do you think about either:

- setting the env vars in BUILD (like MODE)?
- passing cargo etc as `runnable_dependencies` to let pants manage `PATH`? We may have to make `file` runnable, somehow, for this, which seems like a good change

Reply:
The only reason we need a script is because we can't pass PATH as an extra_env_var and have it merged (we can't use $PATH, either as anything we set gets overriden by the binary shims path.

If we were to change the target definition to allow for PATH merging (bikeshed on how, I have an idea or two) we'd just set the PATH in the target's extra_env_vars and be done with it (this is also a comment in the script itself 😉)

---

As for the runnable part, we can't because these aren't single files. There's a handful of ways to fix this, but since we're dogfooding we can't fix it AND make this change in one swoop. We have to be able to compile the engine with what we have first.

So there's options, but this is how to get it working with what we have. (This is huge though. It means there's gaps in what we provide. And we only see that through dogfooding)